src/EventSubscriber/XLocationIdResponseSubscriber.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace EzSystems\PlatformHttpCacheBundle\EventSubscriber;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+
+/**
+ * Rewrites the X-Location-Id HTTP header.
+ *
+ * This is a BC layer for custom controllers (including REST server) still
+ * using X-Location-Id header which is now deprecated. For
+ * full value of tagging, see docs/using_tags.md for how to take advantage of the
+ * system.
+ */
+class XLocationIdResponseSubscriber implements EventSubscriberInterface
+{
+    /**
+     * @var string
+     */
+    private $tagHeader;
+
+    /**
+     * @var LoggerInterface|null
+     */
+    private $logger;
+
+    public function __construct($tagHeader, LoggerInterface $logger = null)
+    {
+        $this->tagHeader = $tagHeader;
+        $this->logger = $logger;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [KernelEvents::RESPONSE => ['rewriteCacheHeader', -5]];
+    }
+
+    public function rewriteCacheHeader(FilterResponseEvent $event)
+    {
+        $response = $event->getResponse();
+        if (!$response->headers->has('X-Location-Id')) {
+            return;
+        }
+
+        if ($this->logger instanceof LoggerInterface) {
+            $this->logger->notice(
+                'X-Location-Id rewritten to use ezplatform-http-cache tag header, see \<bundle\>/docs/using_tags.md for more info'
```

Comment:
What about a good old `E_USER_DEPRECATED` rather than a notice ? Those are way more visible by developers. As long as our own code doesn't throw it ;)